File: leetcode3/염혜정/3827. Count Monobit Integers.java
// 0을 제외하고 1, 11, 111, 1111... (2^1 - 1, 2^2 - 1, 2^3 -1)
// o(log n)

class Solution {
    public int countMonobit(int n) {
        int cnt = 1; // 0은 무조건 포함
        int num = 1;
        while (num <= n) {
            cnt++;
            num = num * 2 + 1;
        }
        return cnt;
    }
}

File: leetcode3/최원준/3827. Count Monobit Integers.py
class Solution:
    def countMonobit(self, n: int) -> int:
        def is_monobit(num):
            binary = bin(num)[2:]
            base = binary[0]
            for b in binary:
                if b != base:
                    return False
            return True

        ans = 0
        for i in range(n+1):
            ans += is_monobit(i)
        return ans

        
        
